components/common/src/kafka_client.py

```diff
@@ -0,0 +1,225 @@
+NEXT_MSG_TIMEOUT = 60
+
+
+def build_configuration(
+    self,
+    bootstrap_servers,
+    client_id,
+    logger,
+    security_protocol,
+    sasl_mechanisms,
+    group_id=None,
+    username=None,
+    password=None,
+    ssl_ca=None,
+    ssl_key=None,
+    ssl_certificate=None,
+    config_params=None,
+    debug=False,
+):
+    configuration = {
+        "bootstrap.servers": bootstrap_servers,
+        "group.id": group_id,
+        "client.id": client_id,
+        "security.protocol": security_protocol,
+        "sasl.mechanisms": sasl_mechanisms,
+        "sasl.username": username,
+        "sasl.password": password,
+        "ssl.ca.location": self._create_temp_file(ssl_ca),
+        "ssl.key.location": self._create_temp_file(ssl_key),
+        "ssl.certificate.location": self._create_temp_file(ssl_certificate),
+        "logger": logger,
+    }
+    if debug:
+        configuration["debug"] = "consumer, broker"
+
+    if group_id:  # if consumer add following params
+        configuration.update(
+            {
+                "session.timeout.ms": 6000,
+                # we are controlling offset ourselves, by default start from start
+                "auto.offset.reset": "earliest",
+                "enable.auto.commit": True,
+            }
+        )
+
+    if config_params:
+        configuration.update(config_params)
+
+    # kafka config can't handle None or "" values
+    return {key: value for key, value in configuration.items() if value is not None}
+
+
+class KafkaProducer:
+    def __init__(
+        self,
+        bootstrap_servers,
+        client_id,
+        logger,
+        security_protocol,
+        sasl_mechanisms,
+        username=None,
+        password=None,
+        ssl_ca=None,
+        ssl_key=None,
+        ssl_certificate=None,
+        config_params=None,
+        debug=False,
+    ):
+        configuration = build_configuration(
+            self,
+            bootstrap_servers=bootstrap_servers,
+            client_id=client_id,
+            logger=logger,
+            security_protocol=security_protocol,
+            sasl_mechanisms=sasl_mechanisms,
+            username=username,
+            password=password,
+            ssl_ca=ssl_ca,
+            ssl_key=ssl_key,
+            ssl_certificate=ssl_certificate,
+            config_params=config_params,
+            debug=debug,
+        )
+
+        self.producer = Producer(**configuration)
+
+    def list_topics(self):
+        return self.producer.list_topics(timeout=60).topics
+
+    def produce_message(self, topic, key, value):
+        self.producer.produce(topic=topic, key=key, value=value)
+        self.producer.flush()
+
+    @staticmethod
+    def _create_temp_file(content, suffix=".pem"):
+        if content:
+            temp_file = tempfile.NamedTemporaryFile(suffix=suffix, delete=False)
+            temp_file.write(content.encode())
+            temp_file.close()
+            return temp_file.name
```

Comment:
This is precisely the type of method which should be extracted into some utility module. 

1. the code is used exactly as is in consumer
2. although the name suggest it's universal, the  default suffix value kinda gives it a specific purpose.

- It could just be a function. 
- It should use resource manager 
- The behaviour is poorly defined (partially due to lack of documentation) -- e.g. I wouldn't expect this to NOT create the file if I want to write an empty string into it (what's worse is that the return value differs based on whehter content is truthy or not